packages/api-derive/src/balances/account.ts

```diff
@@ -55,7 +55,13 @@ function queryBalancesFree (api: ApiInterfaceRx, accountId: AccountId): Observab
 }
 
 function queryBalancesAccount (api: ApiInterfaceRx, accountId: AccountId, modules: string[] = ['balances']): Observable<Result> {
-  const balances = modules.map((m): QueryableStorageMultiArg<'rxjs'> => [api.query[m].account, accountId]);
+  const balances = modules.map(
+    (m): QueryableStorageMultiArg<'rxjs'> => [
+      (api.derive[m as 'balances'] as unknown as ApiInterfaceRx['query']['balances'])?.account ?? api.query[m].account,
```

Comment:
Ummm.... now after merge, I don't see how this works - this points to an existing derive entry, so basically will always have a circular ref to itself. So it will never do the underlying query.

Reply:
... need more coffee first thing in the morning.

Reply:
But does not _api_ object have a reference the instance that called at runtime? That way it could be used to call our custom derives. The trick for multiple currencies in our chain is that they are resolved by a single pallet, that's why I thought that a good idea was to create derives for all our currencies to bind account ballance parameter, eg. `api.derive.btc.account = (accountId) => api.query.balances.account(accountId, "Btc")`

Reply:
The idea is good - for the derive through since there is an exitsing `api.derive.balances.account` it will fallback to that one in all cases. (Since that is the default for chains without any instances, only the `balances` being queried). the derive gets checked first, it is always available and will be used, no queries.

So it is probably just as simple as a rename here, something like `api.derive.balances.customAccount` & `api.derive.balances.customLocks` (aligning these two and ensuring no overrides of existing).

Reply:
thanks, will update now

Reply:
Sorry about this - I read, thought, read again and then on merge saw it. So my lack of being awake made this slightly a longer process.

Reply:
https://github.com/polkadot-js/api/pull/3248